docs/en/setup/envoy/metrics_service_setting.md

```diff
@@ -39,6 +39,17 @@ A more complete static configuration, can be observed [here](config.yaml).
 
 Note that Envoy can also be configured dynamically through [xDS Protocol](https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md).
 
+**Attention**: Only use this when Envoy is under Istio's control, because SkyWalking needs to parse the service name and service instance name from the metadata that is injected by Istio. However, if you want to use this without Istio, you need to inject the metadata yourself like this:
```

Comment:
cc @hanahmily Due to share the same node id mechanism, even we don't require header exchange in the metrics service analysis. We will still depend on Istio control. We fetch the name based on that.